test/errorHandling.test.tsx

```diff
@@ -0,0 +1,265 @@
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        isLoading: false,
+        isError: false,
+        isSuccess: true,
+        data: { value: 'success' },
+      })
+    );
+  });
+
+  test('error', async () => {
+    server.use(
+      rest.get('http://example.com/query', (_, res, ctx) => res(ctx.status(500), ctx.json({ value: 'error' })))
+    );
+
+    const { result } = renderHook(() => api.useQueryQuery({}), { wrapper: storeRef.wrapper });
+
+    await hookWaitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        isLoading: false,
+        isError: true,
+        isSuccess: false,
+        error: { status: 500, data: { value: 'error' } },
+      })
+    );
+  });
+
+  test('success -> error', async () => {
+    server.use(rest.get('http://example.com/query', (_, res, ctx) => res(ctx.json({ value: 'success' }))));
+
+    const { result } = renderHook(() => api.useQueryQuery({}), { wrapper: storeRef.wrapper });
+
+    await hookWaitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        isLoading: false,
+        isError: false,
+        isSuccess: true,
+        data: { value: 'success' },
+      })
+    );
+
+    server.use(
+      rest.get('http://example.com/query', (_, res, ctx) => res.once(ctx.status(500), ctx.json({ value: 'error' })))
+    );
+
+    act(result.current.refetch);
+
+    await hookWaitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        isLoading: false,
+        isError: true,
+        isSuccess: false,
+        error: { status: 500, data: { value: 'error' } },
+        // last data will stay available
+        data: { value: 'success' },
+      })
+    );
+  });
+
+  test('error -> success', async () => {
+    server.use(rest.get('http://example.com/query', (_, res, ctx) => res(ctx.json({ value: 'success' }))));
+    server.use(
+      rest.get('http://example.com/query', (_, res, ctx) => res.once(ctx.status(500), ctx.json({ value: 'error' })))
+    );
+
+    const { result } = renderHook(() => api.useQueryQuery({}), { wrapper: storeRef.wrapper });
+
+    await hookWaitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        isLoading: false,
+        isError: true,
+        isSuccess: false,
+        error: { status: 500, data: { value: 'error' } },
+      })
+    );
+
+    act(result.current.refetch);
+
+    await hookWaitFor(() => expect(result.current.isFetching).toBeFalsy());
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        isLoading: false,
+        isError: false,
+        isSuccess: true,
+        data: { value: 'success' },
+        // do we want to keep this behaviour?
+        // atm., error is not cleared out
```

Comment:
we should probably clear this out when going from isError -> isSuccess, right?

Reply:
in general, yes :)

Reply:
added this in :)